docs/lob_guide/src/asciidoc/_chapters/work_with_lob.adoc

```diff
@@ -0,0 +1,738 @@
+
+* Extract lob data into a user specified buffer
++
+Extract from a LOB column into a target user buffer of a specified size. 
++
+Trafodion engine will retrieve exactly the amount of requested data. 
++
+The user/application will be responsible for redriving the extract until EOD. 
++
+The extracted buffer data can then be written to files or another location as the application chooses. 
+
+[#extract lob data into a file for a given lob handle]
+=== Extract Lob Data into a File for a Given Lob Handle 
+
+[#syntax]
+==== Syntax
+
+```
+EXTRACT LOBTOFILE (LOB 'lob handle as quoted string', 'filename URI format' [OPTION]) 
+```
+
+```
+OPTION is:                                         
+[, TRUNCATE]
+[, CREATE, TRUNCATE]
+[, APPEND]   
+[, CREATE, APPEND]
+```
+
+[#semantics]
+==== Semantics
+
+* `TRUNCATE`
++
+If the target file exists, Trafodion will truncate and write to it.
++
+If the target file does not exist, an error will be raised.
+
+* `CREATE, TRUNCATE`
++
+If the target file exists, Trafodion will truncate and write to it. 
++
+If the target file does not exist, create a table, then write to it. 
+
+* `APPEND`
++
+If the target file exists, Trafodion will append to it. 
++
+If the target file does not exist, an error will be raised.
+
+* `CREATE, APPEND`
++
+If the target file exists, Trafodion will append to it. 
++
+If the target file does not exist, create a table, then write to it. 
+
+[#considerations]
+==== Considerations
+
+If the target file exists, the `OPTION` must be specified, or else an error will be raised. This is the default behavior.
+
+If the target file does not exist, you can create a target file by specifying the `OPTION`. 
+
+[#examples]
+==== Examples
+
+* This example extracts LOB to the tlob130_txt1.txt:
+
++
+```
+EXTRACT LOBTOFILE (LOB 'LOBH00000200010520117997292583625519884121437206093184618212317486018305654020"TRAFODION"."LOB130"', 'tlob130_txt1.txt');
+Success. Targetfile:tlob130_txt1.txt  Length: 19
+```
+
+* This example extracts LOB to the tlob130_deep.jpg:
++
+```
+EXTRACT LOBTOFILE (LOB 'LOBH00000200010520117997292583681719884121437210516812518212317486062586654020"TRAFODION"."LOB130"', 
+Success. Targetfile:tlob130_deep.jpg  Length: 159018
+```
+
+[#extract lob data into a user specified buffer]
+=== Extract Lob Data into a User Specified Buffer
+
+Extract from a LOB column into a target user buffer of a specified size. Trafodion engine will retrieve exactly the amount of requested data. The user/application will be responsible for redriving the extract until EOD. Then the extracted buffer data can be written to files or another location as the application chooses. 
+
+Extract LOB data into a user specified buffer like a cursor until EOD is returned. For this method, the user specifies an input buffer and specifies the input length in an in/out variable. 
+
+For each execution of the extract, Trafodion will return SUCCESS, ERROR or EOD. 

```

Comment:
Are SUCCESS, ERROR and EOD literals in some programming language (such as Java or C++)? Or is it a string that is returned by the Trafodion engine? Or something else? (The sqlci examples below don't mention SUCCESS, ERROR or EOD.) Seems like some additional explanation or examples are needed.

Reply:
Suggestion from Sandhya:

Change 
"For each execution of the extract, Trafodion will return SUCCESS, ERROR or EOD. 
For the cases of SUCCESS or EOD, a length will also be returned to the user, so the user knows exactly how much data was actually extracted and returned. "

to

"Each successful execution of the extract will also return the length in the parameter specified by the caller, so the user knows how much data was actually extracted and returned. "